lib/openshift/machine_set.rb

```diff
@@ -45,5 +45,11 @@ def is_windows_machinesets?(user: nil, cached: true, quiet: false)
       rr = raw_resource(user: user, cached: cached, quiet: quiet)
       rr.dig('spec', 'template', 'metadata', 'labels','machine.openshift.io/os-id') == "Windows"
     end 
+    def machineset_flavor(user: nil, cached: true, quiet: false)
+      rr = raw_resource(user: user, cached: cached, quiet: quiet)
+      ms_provider_spec = rr.dig('spec', 'template', 'spec', 'providerSpec', 'value')
+      flavor = ms_provider_spec['instanceType'] || ms_provider_spec['vmSize'] || ms_provider_spec['machineType'] || ms_provider_spec['flavor']
```

Comment:
@yanpzhan so the `flavor` is determined by any of the spec?  Does it mean only one can be set at a time?

Reply:
yes, only one option in a cluster according to different platforms.